src/config.rs

```diff
@@ -28,8 +28,43 @@ pub struct Config {
     pub flatpak: FlatpakConfig,
     #[serde_inline_default(Config::default().vscode)]
     pub vscode: VsCodeConfig,
+    pub yarn: YarnConfig,
 }
 
+macro_rules! apply_configs {
+    ($macro:ident) => {
+        $macro! {
+        (CargoConfig, cargo),
+        (YarnConfig, yarn),
+        (ArchConfig, arch),
+        (VsCodeConfig, vscode),
+        (FlatpakConfig, flatpak) }
+    };
+}
+
+macro_rules! impl_asRef {
+    ($(($config_type:ident, $config:ident)),*) => {
+        $(
+        impl AsRef<$config_type> for Config {
+            fn as_ref(&self) -> &$config_type{
+            &self.$config
+        }
+        })*
+    };
+}
+
+apply_configs!(impl_asRef);
+
+impl AsRef<()> for Config {
+    fn as_ref(&self) -> &() {
+        &()
+    }
+}
+#[serde_inline_default]
+#[derive(Debug, Serialize, Deserialize, Default)]
+#[serde(deny_unknown_fields)]
+pub struct YarnConfig {}
```

Comment:
Hmm, maybe it would make more sense if each backends' config was in each backends' file just below their options structs. Puts the object closer to where it is used.

Reply:
I agree, will adapt